server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -433,6 +436,11 @@ protected BlobStoreRepository(
             this::doSnapshotShard,
             this::snapshotFile
         );
+        snapshotDeleteTaskRunner = new ThrottledTaskRunner(
+            "DeleteSnapshotTaskRunner",
+            Math.max(1, threadPool.info(ThreadPool.Names.SNAPSHOT_META).getMax() / 3),
```

Comment:
Not sure what this value should be. I'm guessing we want it to be a fraction of the `SNAPSHOT_META` size.

Reply:
This seems reasonable to me. My intuition is that deletes can sometimes take a while (when done in bulk) so we don't want to use up all the meta threads here. The meta pool is sized to `3*$PROCS` so this means we might have at most `$PROCS` delete tasks waiting.

AIUI we don't parallelise the deletes today much: each snapshot delete does all its blob deletes on a single thread, as does each snapshot finalization (which only deletes metadata).